src/Type/Schema.php

```diff
@@ -411,6 +416,8 @@ private function getPossibleTypeMap()
         if ($this->possibleTypeMap === null) {
             $this->possibleTypeMap = [];
             foreach ($this->getTypeMap() as $type) {
+                $type = Type::resolveLazyType($type);
```

Comment:
`getTypeMap` should also always return a map of fully resolved types. This should be removed. Any errors would indicate that `getTypeMap` implementation must be adjusted

Reply:
Right again! removed...